smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -49,6 +51,11 @@ contract MixinKeys is
   // Addresses are never removed by design to avoid abuses around referals
   address[] public owners;
 
+  // A given key has both an owner and a manager.
+  // The owner and manager may be the same address
+  // Each key can have at most 1 keyManager, but may have no manager.
```

Comment:
What does no manager mean? Does that imply the keyOwner is the manager at that time? I think that could make sense but the current implementation I think effectively disables the key if the manager is 0

Reply:
Also... not sure on this. But should keyManager be a field of the Key struct?

Reply:
I don't feel strongly about this either way, but I thought we had discussed and decided that address(0) was a valid keyManager address.
The key would still work, but not be transferrable/shareable.

Reply:
Okay that's a fair use case. Just ensure there's a comment to that effect - maybe on the setter. Something like 'setting to 0 disables transfers for this key'.

Hmm begs a question. What if I buy with CC, Unlock is the key manager. They set the keyManager to 0. A few months later I buy again - this time with crypto. ..but I still cannot transfer? I'll never get those permissions back?

Reply:
good point.of course, Unlock should probably have set themselves as the keyManager, but maybe supporting the zero address as a keyManager is overly complicated for the limited value it brings.

Reply:
Maybe on purchase, if the user does not have a key or the key is expired ATM - they become the keyManager.  e.g. if the first purchase was with CC and then expired before we updated the keyManager - it seems that on next purchase the buyer reclaim that role.

Reply:
Yeah that makes sense. Also, just saw this:
>Also... not sure on this. But should keyManager be a field of the Key struct?

Reply:
Yea I don't really know the tradeoffs. Begs the question - should we have a Key struct at all? And does the answer change knowing Solidity 0.6 is coming which includes abiV2 (which means we can publicly expose structs)

Reply:
Can we make that a new issue to track it? we can always refactor if we decide to change things up.

Reply:
>Maybe on purchase, if the user does not have a key or the key is expired ATM - they become the keyManager. e.g. if the first purchase was with CC and then expired before we updated the keyManager - it seems that on next purchase the buyer reclaim that role.

this has now been fixed.